### PR TITLE
✨Check if the Control Plane is initialized before generating JoinConfigurations

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -212,6 +207,12 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 			}
 		}
 
+		// If the cluster infrastructure is not ready and the spec doesn't have a ControlPlaneEndpoint, requeue.
+		if !cluster.Status.InfrastructureReady && config.Spec.ClusterConfiguration.ControlPlaneEndpoint == "" {
```
Comment (detiber) [inline controllers/kubeadmconfig_controller.go]: Is the vsphere provider setting ControlPlaneEndpoint prior to InfrastructureReady? What determines InfrastructureReady for that provider?

**Comment by detiber** [comment]
Does this resolve https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/125?

**Comment by akutz** [comment]
Hi @vincepri,

Why was this PR closed?

**Comment by vincepri** [comment]
@akutz See https://github.com/kubernetes-sigs/cluster-api/pull/1345

**Comment by akutz** [comment]
Thanks. Maybe in the future when closing an issue in favor of a different solution, add a remark like "closing in favor of ..." or "replacing with ...".

**Pull request closed without merging**